dimod/include/dimod/quadratic_model.h

```diff
@@ -0,0 +1,686 @@
+     * last element.
+     */
+    void emplace_back(index_type v, bias_type bias) {
+        neighbors.push_back(v);
+        quadratic_biases.push_back(bias);
+    }
+
+    /**
+     * Erase an element from the neighborhood.
+     *
+     * Returns the number of element removed, either 0 or 1.
+     */
+    size_type erase(index_type v) {
+        auto it = std::lower_bound(neighbors.begin(), neighbors.end(), v);
+        if (it != neighbors.end() && (*it) == v) {
+            // is there to erase
+            size_type idx = std::distance(neighbors.begin(), it);
+
+            neighbors.erase(it);
+            quadratic_biases.erase(quadratic_biases.begin() + idx);
+
+            return 1;
+        } else {
+            return 0;
+        }
+    }
+
+    /// Erase elements from the neighborhood.
+    void erase(iterator first, iterator last) {
+        quadratic_biases.erase(first.bias_it_, last.bias_it_);
+        neighbors.erase(first.neighbor_it_, last.neighbor_it_);
+    }
+
+    /// Return an iterator to the first element that does not come before `v`.
+    iterator lower_bound(index_type v) {
+        auto it = std::lower_bound(neighbors.begin(), neighbors.end(), v);
+        return iterator{it, quadratic_biases.begin() +
+                                    std::distance(neighbors.begin(), it)};
+    }
+
+    /**
+     * Return the bias at neighbor `v` or the default value.
+     *
+     * Return the bias of `v` if `v` is in the neighborhood, otherwise return
+     * the `value` provided without inserting `v`.
+     */
+    bias_type get(index_type v, bias_type value = 0) const {
+        auto it = std::lower_bound(neighbors.begin(), neighbors.end(), v);
+        size_type idx = std::distance(neighbors.begin(), it);
+        if (it != neighbors.end() && (*it) == v) {
+            // it exists
+            return quadratic_biases[idx];
+        } else {
+            // it doesn't exist
+            return value;
+        }
+    }
+
+    /// Return the size of the neighborhood.
+    size_type size() const { return neighbors.size(); }
+
+    /**
+     * Access the bias of `v`.
+     *
+     * If `v` is in the neighborhood, the function returns a reference to its
+     * bias. If `v` is not in the neighborhood, it is inserted and a reference
+     * is returned to its bias.
+     */
+    bias_type& operator[](index_type v) {
+        auto it = std::lower_bound(neighbors.begin(), neighbors.end(), v);
+        size_type idx = std::distance(neighbors.begin(), it);
+        if (it == neighbors.end() || (*it) != v) {
+            // it doesn't exist so insert
+            neighbors.insert(it, v);
+            quadratic_biases.insert(quadratic_biases.begin() + idx, 0);
+        }
+
+        return quadratic_biases[idx];
+    }
+
+ protected:
+    std::vector<neighbor_type> neighbors;
+    std::vector<bias_type> quadratic_biases;
+};
+
+template <class Bias, class Neighbor = std::int64_t>
```

Comment:
I think the default type should be just, int, reasons. 
1. We save a huge amount of memory now that the vectors are not vectors of pairs and structures are not padded, so if we use bias type float using int will save 1/3rd memory and if bias type is double then it samves 1/4rth memory.

2. We will never get close to using that many variables, and by the time we use that many variables we have to change the algorithms we use for adding biases together since the drift will be too large to handle. 

3. Moving to int from int_64t will not cause any precision bug until and unless we somehow use it in bias computation - something that I did and suffered lol.